rfcs/rfc0020.md

```diff
@@ -0,0 +1,116 @@
+
+A new feature of the Perl 5.18 release was the ability to create lexically-named subroutines, whose names are accessible within some lexical scope but not visible in the package namespace, and thus not visible to callers from outside the code, even via method reflection. Since then, the ability has not been further expanded on by Perl core. While a few CPAN modules have tried to implement the ability, the idea does not seem to have caught on in the majority of cases.
+
+The `builtin` module added in Perl 5.36 was the first core module to export its named functions lexically into the calling scope, rather than symbolically into the caller's package. It aims to set a new standard for better-behaved module exporting, but is currently handled by its own internal implementation which is not yet exposed in a way that other modules can make use of.
+
+## Rationale
+
+By providing a mechanism by which other modules written in pure perl code can take advantage of lexical export themselves, we hope that more will begin to take advantage of this ability. This will lead to modules that behave more politely in the face of the growing collection of modules written in object-oriented style. In addition, as addressed by the "Future Scope" section, we hope this leads on to the creation of better Exporter-like modules or other core perl abilities to take advantage of lexical export.
+
+## Specification
+
+A new function in the `builtin` package, called `lexically_export`. This function takes an even-sized name/value list of pairs. It does not return anything.
+
+```perl
+builtin::lexically_export $name1, $value1, $name2, $value2, ...;
+```
+
+Each name/value pair acts independently. The name gives the new name to be created in the scope being compiled, and the value is a reference to the item the name should have.
+
+Four kinds of items are supported:
+
++ If the value is a `CODE` reference, a new named subroutine will be created in the scope. The corresponding name argument may optionally begin with a `&` sigil.
+
++ If the name is a `SCALAR` reference, a new scalar variable will be created in the scope. The corresponding name argument must begin with a `$` sigil.
+
++ If the name is an `ARRAY` reference, a new array variable will be created in the scope. The corresponding name argument must begin with a `@` sigil.
+
++ If the name is a `HASH` reference, a new hash variable will be created in the scope. The corresponding name argument must begin with a `%` sigil.
```

Comment:
Shouldn't these all be "If the value is a ... reference" rather than "If the *name* is a ... reference"?